x/merkledb/maybe.go

```diff
@@ -35,10 +41,18 @@ func (m Maybe[T]) IsNothing() bool {
 }
 
 // Returns the value of [m].
+// If [m.IsNothing()], returns the zero value of type T.
 func (m Maybe[T]) Value() T {
 	return m.value
 }
 
+func (m Maybe[T]) String() string {
+	if !m.hasValue {
+		return fmt.Sprintf("Nothing[%T]", m.value)
+	}
+	return fmt.Sprintf("Some[%T]{%v}", m.value, m.value)
+}
```

Comment:
I felt like this was a reasonable string representation but if other people have ideas, I'm all ears